### DISCUSSION TITLE
Mutually Exclusive

### DISCUSSION THREAD

**Discussion by bameyrick** [ONSdigital/design-system#29] [Feedback]
Use this place to discuss the [mutually exclusive component](https://ons-design-system.netlify.com/components/mutually-exclusive) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by bameyrick** [comment]
When an option is toggled on or off programatically, a change event should be fired so that any JavaScript listening to change events on the options is notified of the change.

**Comment by bameyrick** [comment]
As discussed with @will-madebymoon, we should remove the mutually exclusive select option, as this functionality can be achieved with a standard select.

**Comment by bameyrick** [comment]
Mutually exclusive does not work when there is no legend supplied for checkboxes, e.g. when wrapping the field in a question component.

**Comment by bameyrick** [comment]
Since updating fields to use p tags (#417) the mutually exclusive checkbox has broken:

![Screenshot 2019-06-13 at 10 40 10](https://user-images.githubusercontent.com/8447446/59423308-cedc5580-8dc9-11e9-9235-a36d941c17f9.png)
![Screenshot 2019-06-13 at 10 40 21](https://user-images.githubusercontent.com/8447446/59423309-cf74ec00-8dc9-11e9-9182-362cfcb50d15.png)

**Comment by bameyrick** [comment]
When used within a question component, the mutually exclusive checkbox is not the correct width

![Screenshot 2019-06-13 at 11 48 04](https://user-images.githubusercontent.com/8447446/59426895-41046880-8dd1-11e9-9ddc-7ffbc075c78c.png)

**Comment by bameyrick** [comment]
The layout of mutually exclusive is still broken but only on singular "input-type" inputs like the currency input

![Screenshot 2019-06-13 at 15 14 50](https://user-images.githubusercontent.com/8447446/59440048-3a381e80-8dee-11e9-97c7-99a87297f3b1.png)

**Comment by bameyrick** [comment]
Theres not enough space between the textarea character count and the mutually exclusive option:

![Screenshot 2019-07-01 at 14 27 59](https://user-images.githubusercontent.com/8447446/60440336-85cf3280-9c0c-11e9-9cc9-74ee2f7b9aa7.png)

**Comment by bameyrick** [comment]
The mutually exclusive tests aren't clearing their templates after each test which could pollute other tests.

**Comment by jrbarnes9** [comment]
Can't add custom attributes to a mutually exclusive checkbox

**Comment by jrbarnes9** [comment]
Inconsistent vertical spacing between duration and date field groups and mutually exclusive checkbox

![Screenshot 2021-06-14 at 16 11 18](https://user-images.githubusercontent.com/43346934/121915822-7bd86880-cd2b-11eb-870a-087f333fc865.png)

**Reply by jrbarnes9** [reply]
Raised #1556

**Comment by jrbarnes9** [comment]
Documentation doesn't include how to validate if no js

**Reply by jrbarnes9** [reply]
Raised #1766

**Comment by Al-Davies** [comment]
#Mutually Exclusive User Research
tested as part of eQ usability testing 7 March 2022

* Participants clear that if they selected a top answer, then subsequently selected an answer from the ‘or’ section, their other answers would be removed when testing with Multiple Exclusive Answer Options
* Participants clear when only one mutually exclusive answer present, however this became less clear when the question asked for contact details, and the OR read 'I do not wish to be contacted' - participants related this to you are collecting my details but I don't want you to contact me in any way. It was less obvious in this scenario that the 'or' answer would clear previously entered contact details

**Reply by jrbarnes9** [reply]
Raised #2155
